pkg/apis/core/validation/validation_test.go

```diff
@@ -4118,7 +4163,33 @@ func TestValidateEnv(t *testing.T) {
 					},
 				},
 			}},
-			expectedError: `[0].valueFrom.fieldRef.fieldPath: Unsupported value: "metadata.annotations": supported values: "metadata.name", "metadata.namespace", "metadata.uid", "spec.nodeName", "spec.serviceAccountName", "status.hostIP", "status.podIP"`,
+			expectedError: `[0].valueFrom.fieldRef.fieldPath: Unsupported value: "metadata.annotations": supported values: "metadata.annotations[]", "metadata.labels[]", "metadata.name", "metadata.namespace", "metadata.uid", "spec.nodeName", "spec.serviceAccountName", "status.hostIP", "status.podIP"`,
```

Comment:
it's odd that `FieldPath: "metadata.annotations[]"` results in an error that tells you `metadata.annotations[]` is a supported value

Reply:
Because `"metadata.annotations[]"` is invalid, but we should document the usage of field path in field selector and Downward API.